tests/ShrinkingGC.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2025 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/*
+ * @test
+ * @summary Verifies that heap is shrunk in size
+ *          by a special GC invocation through SystemUtils.shrinkingGC() API
+ * @run main/othervm -XX:JbrShrinkingGcMaxHeapFreeRatio=50 -XX:+UseG1GC -Xmx2G pass
+ * @run main/othervm -XX:JbrShrinkingGcMaxHeapFreeRatio=50 -XX:+UseZGC -Xmx2G
+ * @run main/othervm -XX:JbrShrinkingGcMaxHeapFreeRatio=50 -XX:+UseShenandoahGC -Xmx2G
+ * @run main/othervm -XX:JbrShrinkingGcMaxHeapFreeRatio=50 -XX:+UseParallelGC -Xmx2G
+ * @run main/othervm -XX:JbrShrinkingGcMaxHeapFreeRatio=50 -XX:+UseSerialGC -Xmx2G
+ * @run main/othervm -XX:JbrShrinkingGcMaxHeapFreeRatio=50 -XX:+UnlockExperimentalVMOptions -XX:+UseEpsilonGC -Xmx2G
+ * @run main/othervm -XX:+UseG1GC -Xmx2G
+ * @run main/othervm -XX:+UseZGC -Xmx2G
+ * @run main/othervm -XX:+UseShenandoahGC -Xmx2G
+ * @run main/othervm -XX:+UseParallelGC -Xmx2G
+ * @run main/othervm -XX:+UseSerialGC -Xmx2G
+ * @run main/othervm -XX:+UnlockExperimentalVMOptions -XX:+UseEpsilonGC -Xmx2G
+ */
+
+import com.jetbrains.Extensions;
+import com.jetbrains.JBR;
+
+import java.nio.ByteBuffer;
+
+public class ShrinkingGC {
+    private static final int ITERATIONS = 10;
+
+    public static void main(String[] args) throws Exception {
+        boolean expectTestToPass = args.length > 0 && "pass".equals(args[0]);
+
+        ByteBuffer retained500m = ByteBuffer.allocate(512 * 1024 * 1024);
```

Comment:
If you actually need to retain these 500M for the duration of the test, add some code that makes use of the variable (like `System.out.println(retained500m);`)somewhere towards the end.